ksp_plugin/vessel.cpp

```diff
@@ -334,20 +334,38 @@ Vessel::prediction_adaptive_step_parameters() const {
 }
 
 bool Vessel::has_flight_plan() const {
-  return has_deserialized_flight_plan() ||
-         std::holds_alternative<serialization::FlightPlan>(flight_plan_);
+  return !flight_plans_.empty();
+}
+
+int Vessel::flight_plan_count() const {
+  return flight_plans_.size();
+}
+
+int Vessel::selected_flight_plan() const {
+  return selected_flight_plan_;
+}
+
+void Vessel::SelectFlightPlan(int index) {
+  CHECK_GE(index, 0);
+  CHECK_LT(index, flight_plan_count());
+  selected_flight_plan_ = index;
 }
 
 FlightPlan& Vessel::flight_plan() const {
   CHECK(has_deserialized_flight_plan());
-  auto& flight_plan = *std::get<std::unique_ptr<FlightPlan>>(flight_plan_);
+  auto& flight_plan = *std::get<not_null<std::unique_ptr<FlightPlan>>>(
+      flight_plans_[selected_flight_plan_]);
   return flight_plan;
 }
 
 void Vessel::ReadFlightPlanFromMessage() {
-  if (std::holds_alternative<serialization::FlightPlan>(flight_plan_)) {
-    auto const& message = std::get<serialization::FlightPlan>(flight_plan_);
-    flight_plan_ = FlightPlan::ReadFromMessage(message, ephemeris_);
+  if (!flight_plans_.empty() &&
+      std::holds_alternative<serialization::FlightPlan>(
+          flight_plans_[selected_flight_plan_])) {
```

Comment:
A helper function to access the currently selected flight plan would be useful, this expression shows up everywhere.